internal/controllers/topology/cluster/reconcile_state.go

```diff
@@ -403,21 +403,21 @@ func (r *Reconciler) createMachineDeployment(ctx context.Context, cluster *clust
 		cluster: cluster,
 		desired: md.InfrastructureMachineTemplate,
 	}); err != nil {
-		return errors.Wrapf(err, "failed to create %s", tlog.KObj{Obj: md.Object})
+		return errors.Wrapf(createErrorWithoutObjectName(err, md.InfrastructureMachineTemplate), "failed to create %s", md.Object.Kind)
```

Comment:
q: Is it required to call createErrorWithoutObjectName here, given that it is already called inside reconcileReferencedTemplate?
same below

Reply:
Looks like we can just return `err` here (as the full error is already build in the reconcile func in all cases)